synapse/app/admin_cmd.py

```diff
@@ -222,6 +222,19 @@ def write_connections(self, connections: List[JsonDict]) -> None:
             with open(connection_file, "a") as f:
                 print(json.dumps(connection), file=f)
 
+    def write_account_data(
+        self, file_name: str, account_data: Mapping[str, JsonDict]
+    ) -> None:
+        account_data_directory = os.path.join(
+            self.base_directory, "user_data", "account_data"
+        )
+        os.makedirs(account_data_directory, exist_ok=True)
+
+        account_data_file = os.path.join(account_data_directory, file_name)
+
+        with open(account_data_file, "a") as f:
+            print(json.dumps(account_data), file=f)
```

Comment:
This is fine here (we don't expect account_data to be large), but for general interest it might be more efficient to use `json.dump(account_data, f)` (see [here](https://docs.python.org/3/library/json.html#json.dump)) for large `account_data` blobs.

Reply:
Wouldn't that make sense for all functions in this file?

Reply:
Probably! But we don't need to change it here and now.